### DISCUSSION TITLE
Attach chrome debugger to dev server: `astro dev` should take an `--inspect` flag

### DISCUSSION THREAD

**Discussion by mb21** [withastro/roadmap#397] [Proposal]
I assume `npm run dev`, which executes `astro dev`, runs on node.js? If so, it should be possible to attach the Chrome dev tool's debugger to the node process – if that one was started with the `--inspect` flag. ~~That would make it way easier to debug `fetch` calls or step through javascript execution on the server.~~ Actually, unfortunately the Node.js debugger doesn't show network calls, but this could still be useful for stepping through the javascript line by line and using breakpoints.

see https://nodejs.org/en/docs/guides/debugging-getting-started/#enable-inspector and/or https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27

Sure, `console.log` debugging works. But coming from a pure browser-based frontend project, I'm spoiled by the browser's dev tools. Thanks for considering!

**Comment by SongRainRay** [comment]
你好！您的邮件我已收到！（自动回复）

**Comment by florian-lefebvre** [comment]
I don't know if there's a way to forward `--profile` to vite: https://vite.dev/guide/troubleshooting#performance-bottlenecks. Otherwise, you can probably use `node --inspect-brk` and then point to the astro cli bin, like for vite: https://github.com/vitejs/vite/issues/672#issuecomment-756892375
